### ISSUE TITLE
`Programming exercise`: Use primitive booleans instead of objects for the domain class attributes

### ISSUE THREAD

**Issue by b-fein** [ls1intum/Artemis#6079]
### Is your feature request related to a problem?

A few attributes of the [programming exercise domain class](https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java) use the `Boolean` type instead of plain `boolean`.
This results in the caller having to use `Boolean.FALSE.equals(exercise.attribute())` or similar to decide how a `null`-value should be handled. This is error-prone, as it can be easily forgotten in some places to handle this correctly, which results in inconsistent behaviour.

### Describe the solution you'd like

- Step 1: The getter/setter methods of an attribute should take/return plain `boolean`s and the `Booleans.?.equals(…)` comparison should happen inside.
- Step 2: Refactor the callers of the gett/setter to remove the `Booleans.?.equals()` checks there.
- Step 3: Change the attribute type to plain `boolean` and set an appropriate default value for each one. Remove the internal `Booleans.?.equals` checks in the getter.
- Step 4: Repeat for the next attribute.

This should allow refactoring while being able to check in each step that no tests break, i.e. the behaviour stays the same.

**Comment by 1999Markus** [comment]
Hi Benedikt, I just assigned this task to myself. I took the first step that you described for the attribute `publishBuildPlanUrl` but the following line of the `ParticipationIntegrationTest` does not make sense anymore. 
`                assertThat(aProgrammingExercise.isPublishBuildPlanUrl()).isNull();
`
My hunch would be to change this line (after setting a default value) to check for equality to the expected default value. 
What's your opinion on this?

**Comment by b-fein** [comment]
I’d assume that the build plan only gets published in case `publishBuildPlanUrl` was explicitly set to `true` somewhere. Therefore, the intermediate step 1 for the getter in this case should probably be `return Booleans.TRUE.equals(publishBuildPlanUrl)`.

So in this specific case I’d expect that the assertion should now be
```java
assertThat(aProgrammingExercise.isPublishBuildPlanUrl()).isFalse();
```

**Comment by 1999Markus** [comment]
Hi Benedikt, 
I'm a bit stuck because the tests keep failing for a reason I don't understand.

The initialization of the attribute looks like this right now:
`@Column(name = "allow_online_editor", table = "programming_exercise_details")
    private boolean allowOnlineEditor = false;`
 
and its setter method like this:
`public void setAllowOnlineEditor(boolean allowOnlineEditor) {
        this.allowOnlineEditor = allowOnlineEditor;
    }`

However I get the following error:
`java.lang.NoSuchMethodError: 'void de.tum.in.www1.artemis.domain.ProgrammingExercise.setAllowOnlineEditor(java.lang.Boolean)'`

I don't get why the old parameter type is expected here but my hunch is it has something todo with the database.

**Comment by b-fein** [comment]
Hm, that’s difficult to debug with only this message. Is there a branch where I can see the changes? Which test(s) are failing?

**Comment by 1999Markus** [comment]
Yes I'll publish my branch and commit the current changes. The tests are obvs. gonna fail though

**Comment by b-fein** [comment]
Hm, I made the two changes you mentioned above and all tests in the two classes bases on the `ProgrammingExerciseIntegrationTestService` (`ProgrammingExerciseIntegrationJenkinsGitlabTest`/`ProgrammingExerciseIntegrationBambooBitbucketJiraTest`) still pass.

Might that have been caused by some classes not being recompiled so that an old cached version is used again? I’m not sure when/why that would happen, though.

```diff
diff --git a/src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java b/src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java
index 89f6585ac6..6acaf451bb 100644
--- a/src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java
+++ b/src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java
@@ -52,7 +52,7 @@ public class ProgrammingExercise extends Exercise {
     private Boolean publishBuildPlanUrl;
 
     @Column(name = "allow_online_editor", table = "programming_exercise_details")
-    private Boolean allowOnlineEditor;
+    private boolean allowOnlineEditor = false;
 
     @Column(name = "allow_offline_ide", table = "programming_exercise_details")
     private Boolean allowOfflineIde;
@@ -250,7 +250,7 @@ public class ProgrammingExercise extends Exercise {
         return allowOnlineEditor;
     }
 
-    public void setAllowOnlineEditor(Boolean allowOnlineEditor) {
+    public void setAllowOnlineEditor(boolean allowOnlineEditor) {
         this.allowOnlineEditor = allowOnlineEditor;
     }
 
diff --git a/src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java b/src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java
index cc6b5b3854..68604d982b 100644
--- a/src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java
+++ b/src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java
@@ -526,7 +526,6 @@ public class ParticipationResource {
                 programmingExercise.setPublishBuildPlanUrl(null);
                 programmingExercise.setProgrammingLanguage(null);
                 programmingExercise.setPackageName(null);
-                programmingExercise.setAllowOnlineEditor(null);
             }
             else if (exercise instanceof QuizExercise quizExercise) {
                 quizExercise.setQuizQuestions(null);

```

**Comment by 1999Markus** [comment]
Mh okay could you try checking out my branch and running the tests there?
